cellengine/utils/api_client/APIClient.py

```diff
@@ -477,19 +482,25 @@ def delete_gate(
             raise ValueError("Either _id or gid must be specified.")
         self._delete(url)
 
-    def post_gate(
-        self, experiment_id, gate: Dict, create_population=True, as_dict=False
-    ) -> Gate:
-        res = self._post(
-            f"{self.base_url}/experiments/{experiment_id}/gates",
-            json=gate,
-            params={"createPopulation": create_population},
-        )
-        if as_dict:
-            return res
-        if type(res) is list:
-            return [Gate.factory(r) for r in res]
-        return Gate.factory(res)
+    def delete_gates(self, experiment_id: str, ids: List[str]):
+        url = f"{self.base_url}/experiments/{experiment_id}/gates/"
+        [self._delete(url + _id) for _id in ids]
+
+    def _parse_gate_population(self, res: Any) -> Tuple[Gate, Population]:
+        keys = res.keys()
+        if "population" in keys:
+            gate = res["gate"]
+            pop = res["population"]
+        elif "populations" in keys:
+            gate = res["gate"]
+            pop = res["populations"]
+        else:
+            gate = res
+            pop = None
+        return (
+            converter.structure(gate, Gate),
+            converter.structure(pop, Population) if pop else None,
+        )  # type: ignore
```

Comment:
Could this be avoided by changing the return type to `Tuple[Gate, Union[Population, None]]`?